internal/registry/git.go

```diff
@@ -0,0 +1,37 @@
+package registry
+
+import (
+	"bytes"
+	"os"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+func CreateGitCommit(b Buildpack, username string, registryCache Cache) error {
+	_, err := os.Stat(registryCache.Root)
+	if err != nil {
+		if os.IsNotExist(err) {
```

Comment:
You can combine these lines, to make it a bit simpler:
```suggestion
	if err != nil && if os.IsNotExist(err) {
```
You'd have to take out a closing bracket in addition.